lib/gems/pending/util/miq_object_storage.rb

```diff
@@ -45,6 +45,17 @@ def write_single_split_file_for(file_io)
     clear_split_vars
   end
 
+  def write_chunk_proc(destination)
+    # We use a `proc` here instead of `lambda` because we are only concerned
+    # about the first argument, and other arguments (additional ones added by
+    # Excon's response_call signature, for example) are unneeded.
+    #
+    # `lambda do` will do argument checking, while `proc do` won't.
+    proc do |chunk|
```

Comment:
What are `chunk` and `destination` here? Is this some kind of common object?

Reply:
`destination` is the variable passed in from the method arg...

Reply:
`chunk` on the other hand is just the common var name used by both `aws-sdk` and `fog`:

- https://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html#get_object-instance_method
  - (see the "Streaming data to a block" section)
- https://github.com/excon/excon#streaming-responses
  - (`Excon` is the HTTP lib for `fog`, by the same author)

So just reused that terminology.

Reply:
I understand where the variable names come from, I'm just surprised that the objects we're working with quack the same for both s3 and swift so I was curious what kinds of objects these were in those cases.

Reply:
Or are they just byte strings or something?

Reply:
Ah got it. `chunk` is a string and `destination` is an `IO`. Could have just looked it up a bit, sorry.

Reply:
@carbonin yeah, sorry, the usage of this as a `&block` is probably also not making this easier to grok...